### PR TITLE
Scalability guide

### PR DESCRIPTION
**Pull request by rothgar**
*Description of changes:*
Add scalability best practices :partying_face: 

Also updated some karpenter links so they always point to the latest available version.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: content/karpenter/index.md [hunk 1]
```diff
@@ -253,8 +253,8 @@ If you need to run highly available applications, follow general EKS best practi
...
-* Requiring certain kinds of processors or other hardware. See the [Accelerators](https://karpenter.sh/docs/concepts/scheduling/#acceleratorsgpu-resources) section of the Karpenter docs for a podspec example that requires the pod to run on a GPU.
+* Needing to run in availability zones where only particular applications are available. Say, for example, you have pod that has to communicate with another application that runs on an EC2 instance residing in a particular availability zone. If your aim is to reduce cross-AZ traffic in your VPC, you may want to co-locate the pods in the AZ where the EC2 instance is located. This sort of targeting is often accomplished using node selectors. For additional information on [Node selectors](https://karpenter.sh/preview/tasks/scheduling/#node-selectors), please refer to the Kubernetes documentation.
```
Comment (jicowan) [inline content/karpenter/index.md:256]: Is this link right? Can you submit a separate PR for corrections? I'd like to keep this as clean as possible.

**Review by jicowan — COMMENTED** [review]
Can you remove the changes to the other part of this guide from this PR? This PR should only include changes related to Scalability.

**Comment by rothgar** [comment]
I'm pretty sure I pulled out all the other changes. They came in from me failing to rebase.

**Review by jicowan — APPROVED** [review]
/LGTM
